### PR TITLE
App communication docs

### PR DESCRIPTION
**Pull request by krzysztofwolski**
Ref #570

### CODE REVIEW

File: docs/developer/extending/apps/architecture/communication-between-app-and-saleor.mdx [hunk 1]
```diff
@@ -1,3 +1,142 @@
 ---
 title: Communication between App and Saleor
 ---
+
+## Saleor Asynchronous Events
```
Comment (lkostrowski) [inline docs/developer/extending/apps/architecture/communication-between-app-and-saleor.mdx:5]: I think you can use h1?

File: docs/developer/extending/apps/architecture/communication-between-app-and-saleor.mdx [hunk 2]
```diff
@@ -1,3 +1,142 @@
...
+
+In this scenario, the source of the request is a Saleor instance (the worker - to be exact). In contrast to synchronous events, Saleor does not expect an immediate response from the App.
```
Comment (lkostrowski) [inline docs/developer/extending/apps/architecture/communication-between-app-and-saleor.mdx:7]: Can you link to Core docs?

File: docs/developer/extending/apps/architecture/communication-between-app-and-saleor.mdx [hunk 3]
```diff
@@ -1,3 +1,142 @@
...
+```mermaid
+sequenceDiagram
+    participant c as Customer
+    participant s as Saleor API
+    participant w as Saleor Worker
+    participant a as App
+    participant extSales as External sales reporting service
+
+    c ->>+ s: create order
+    s --> w: order created event
```
Comment (lkostrowski) [inline docs/developer/extending/apps/architecture/communication-between-app-and-saleor.mdx:20]: maybe make it explicit:
`s --> w: "ORDER_CREATED" event`

File: docs/developer/extending/apps/architecture/communication-between-app-and-saleor.mdx [hunk 4]
```diff
@@ -1,3 +1,142 @@
...
+    w ->>+ a: send event payload
+    a ->> a: process the payload
+    a ->> extSales: send new order data
+    a ->>- w: confirm task execution
+```
+
+To process webhooks, the App needs publicly accessible endpoints listening for HTTP POST requests. Before requests will be sent, App must [subscribe for events](developer/extending/apps/asynchronous-webhooks.mdx).
+Event delivery will retry if:
+
+- endpoint does not respond before timeout
```
Comment (lkostrowski) [inline docs/developer/extending/apps/architecture/communication-between-app-and-saleor.mdx:32]: What is the timeout? Can we link to docs?

File: docs/developer/extending/apps/architecture/communication-between-app-and-saleor.mdx [hunk 5]
```diff
@@ -1,3 +1,142 @@
...
+
+### Task queue
+
+Example flow of the data for application sending data to reporting service:
+
+```mermaid
+sequenceDiagram
+    participant c as Customer
+    participant s as Saleor API
+    participant q as Queue service - ex SQS
```
Comment (lkostrowski) [inline docs/developer/extending/apps/architecture/communication-between-app-and-saleor.mdx:45]:
```suggestion
    participant q as Queue service - e.g. SQS
```

e.g. = for example

File: docs/developer/extending/apps/architecture/communication-between-app-and-saleor.mdx (hunk 1 above)
Comment (krzysztofwolski) [reply docs/developer/extending/apps/architecture/communication-between-app-and-saleor.mdx:5]: Should be h2. @peelar suggested using it in the docs (metadata docs use it for example)

**Review by lkostrowski — APPROVED** [review]

**Pull request merged**